tests/parts_connection_run_test.py

```diff
@@ -0,0 +1,163 @@
+#
+#  All rights reserved.
+#
+from pathlib import Path
+
+import numpy as np
+from geoh5py import Workspace
+from geoh5py.data import FilenameData
+from geoh5py.objects import Curve, Points
+from geoh5py.ui_json import InputFile
+
+from curve_apps.parts_connection.constants import default_ui_json
+from curve_apps.parts_connection.driver import PartsConnectionDriver
+from curve_apps.parts_connection.params import Parameters
+
+
+def setup_example(workspace: Workspace):
+    with workspace.open(mode="r+"):
+        y_array = np.linspace(0, 50, 10)
+
+        curvey = 5 * np.sin(y_array) + 10
+        oblique = 0.7 * y_array + 20
+        straight = np.ones_like(y_array) * 30
+        randos = np.random.rand(len(y_array)) * 10
+
+        anomalies = [curvey, oblique, straight, randos]
+
+        vertices = np.c_[
+            np.hstack(anomalies),
+            np.tile(y_array, len(anomalies)),
+            np.zeros(len(y_array) * len(anomalies)),
+        ]
+        parts = np.tile(np.arange(len(y_array)), len(anomalies))
+        values = np.r_[
+            np.kron(np.arange(1, 4), np.ones(len(y_array))), np.zeros(len(y_array))
+        ]
+
+        curve = Curve.create(workspace=workspace, vertices=vertices, parts=parts)
+        data = curve.add_data(
+            {
+                "values": {
+                    "values": values.flatten(order="F"),
+                    "value_map": {1: "A", 2: "B", 3: "C", 4: "D"},
+                    "type": "referenced",
+                },
+            }
+        )
+
+    return curve, data
+
+
+def test_driver_curve(tmp_path: Path):
+    workspace = Workspace.create(tmp_path / "test_parts_connection.geoh5")
+
+    curve, data = setup_example(workspace)
+    params = Parameters.instantiate(
+        {
+            "geoh5": workspace,
+            "entity": curve,
+            "data": data,
+            "export_as": "test",
+        }
+    )
+
+    driver = PartsConnectionDriver(params)
+    driver.run()
+
+    with workspace.open():
+        edges = workspace.get_entity("test")[0]
+
+        assert len(edges.cells) == 27
+
+
+def test_driver_points(tmp_path: Path):
+    workspace = Workspace.create(tmp_path / "test_parts_connection.geoh5")
+
+    curve, data = setup_example(workspace)
+
+    with workspace.open():
+        points = Points.create(workspace, vertices=curve.vertices)
+        parts = points.add_data(
+            {
+                "parts": {
+                    "values": curve.parts + 1,
+                },
+            }
+        )
+        new_data = data.copy(parent=points)
+
+    params = Parameters.instantiate(
+        {
+            "geoh5": workspace,
+            "entity": points,
+            "parts": parts,
+            "data": new_data,
+            "export_as": "test",
+        }
+    )
+
+    driver = PartsConnectionDriver(params)
+    driver.run()
+
+    with workspace.open():
+        edges = workspace.get_entity("test")[0]
+
+        assert len(edges.cells) == 27
```

Comment:
isn't checking only for the length a bit light for a test?
we don't expect specific values?